### ISSUE TITLE
Feature request: page no. relative to Zuz and Surah

### ISSUE THREAD

**Issue by abillahbd** [quran/quran_android#1185]
Assalamualaikum brothers! I am memorizing the Quran with this app by the grace of Allah. Already completed 15 zuz! Alhamdulillah!

Please add page number relative to the Zuz at the top of every page. This will help me insha Allah because without this, it's very difficult to remember which page comes after which page. For example: zuz 1 page 1; zuz 1 page 2; zuz 1 page 3... then when the next zuz comes, zuz 2 page 1, zuz 2 page 2, zuz 2 page 3... likewise.

Do this also relative to the surah number. For example: surah 3 page 1, surah 3 page 2, surah 3 page 3... likewise.

Maybe then have an option to turn this paging on/off in the settings.

May Allah reward you with good.

**Comment by murtraja** [comment]
Well, I generally, do a quick calculation in my head to help me understand the safah no. relative to Juz. Generally the Juz starting Safah has a pattern: 1, 22, 42, 62, 82, etc. so given a safah no. I simply get the difference from the previous closest Juz start position. For e.g. given 154, we can estimate that 152 must be halfway, therefore, the relative safah number is 10 + (154-152) + 1 = 13

I do think that this will be useful for memorization, but we must do it without creating any confusion with the actual absolute safah no. One idea that pops up is we might use dot notation for relative numbering like J2.5 will stand for 2nd juz 5th page (or simply 5/20) and S2.5 might represent Surah 2, safah 5

Since the absolute page numbers are centered, the relative page numbers might be displayed on the right or left.

**Comment by abillahbd** [comment]
> Well, I generally, do a quick calculation in my head to help me understand the safah no. relative to Juz. Generally the Juz starting Safah has a pattern: 1, 22, 42, 62, 82, etc. so given a safah no. I simply get the difference from the previous closest Juz start position. For e.g. given 154, we can estimate that 152 must be halfway, therefore, the relative safah number is 10 + (154-152) + 1 = 13
> 
> I do think that this will be useful for memorization, but we must do it without creating any confusion with the actual absolute safah no. One idea that pops up is we might use dot notation for relative numbering like J2.5 will stand for 2nd juz 5th page (or simply 5/20) and S2.5 might represent Surah 2, safah 5
> 
> Since the absolute page numbers are centered, the relative page numbers might be displayed on the right or left.

Yes! Besides the absolute page number can be given a different styling, making it look more prominent among the other numbers. Or maybe just put an option in the settings to turn these requested features on. In this case, setting the features "turned off" by default will solve the confusion problem for those who don't want this feature.

**Comment by ahmedre** [comment]
not completely against adding this as a feature with an option to explicitly enable it.

**Comment by abillahbd** [comment]
> not completely against adding this as a feature with an option to explicitly enable it.

Assalamu Alaikum brother, I see that you are a contributor to this project. Thanks for your kind consideration. This is help a lot truly.

May Allah reward you with good tremendously.
